tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12773,6 +12830,30 @@ HLSLMaxRecordsAttr *ValidateMaxRecordsAttributes(Sema &S, Decl *D,
                          A.getAttributeSpellingListIndex());
 }
 
+HLSLWaveSizeAttr *ValidateWaveSizeAttributes(Sema &S, Decl *D,
```

Comment:
Given the similar name to ValidateWaveSize above, a quick descriptive comment explaining why this function is separate might allay some future confusion.